app/controllers/products_controller.rb

```diff
@@ -78,6 +78,24 @@ def create
     end
   end
 
+  def retire
+    unless @merchant == @product.merchant
+      flash[:status] = :error
+      flash[:result_text] = "Unauthorized user"
+      return redirect_to root_path
+    end
+    @product.status == "active" ? @product.status = "retired" : @product.status = "active"
+    if @product.save
+      flash[:status] = :success
+      flash[:result_text] = "Product status was updated to #{@product.status}."
+    else
+      flash[:status] = :error
+      flash[:result_text] = "Product status couldn't be updated."
+      flash[:messages] = @product.errors.messages
+    end
+    return redirect_back fallback_location: root_path
+  end
+
```

Comment:
Not for now, but when you and Victoria work on refactoring we can look at pulling some of these repeated lines of code and pulling them out into a method.